src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -20,9 +21,10 @@ public MessageReceiverNotifier(IManageMessageReceiverLifeCycle clientEntities, I
             this.settings = settings;
             locksTokensToComplete = new ConcurrentStack<Guid>();
             batchedCompletionCts = new CancellationTokenSource();
+            RefCount = 1;
         }
 
-        bool ShouldReceiveMessages => !stopping || isRunning;
+        bool ShouldReceiveMessages => !stopping;
 
         public bool IsRunning => isRunning;
```

Comment:
I propose we address this when we backport the changes to 7.1, are you OK with that proposal?